analytics/src/analytics/etl/github.py

```diff
@@ -1,38 +1,31 @@
+"""A series of ETL functions to export data from GitHub."""
 import shlex
 import subprocess
+from pathlib import Path
 
-OWNER = "HHS"
-REPO = "simpler-grants-gov"
-SPRINT_PROJECT = 13
 
-
-def pipe_command_to_file(command: str, output_file: str):
-    with open(output_file, "w") as f:
-        subprocess.call(shlex.split(command), stdout=f)
+def pipe_command_output_to_file(command: str, output_file: str) -> None:
+    """Write the command line output to a file."""
+    # make sure the output file's directory exists
+    file_path = Path(output_file)
+    file_path.parent.mkdir(exist_ok=True, parents=True)
+    # invoke the command via a subprocess and write the output to a file
+    with open(output_file, "w", encoding="utf-8") as f:
+        subprocess.call(shlex.split(command), stdout=f)  # noqa: S603
 
 
 def export_project_data(owner: str, project: int, output_file: str) -> None:
-    """Exports and writes GitHub project data to a JSON file"""
+    """Export and write GitHub project data to a JSON file."""
+    print(f"Exporting project data from {owner}/{project} to {output_file}")
     command = f"gh project item-list {project} --format json --owner {owner} -L 1000"
-    pipe_command_to_file(command, output_file)
+    pipe_command_output_to_file(command, output_file)
 
 
 def export_issue_data(owner: str, repo: str, output_file: str) -> None:
-    """Exports and writes GitHub issue data to a JSON file"""
-    command = f"gh issue list --json number,createdAt,closedAt -R {owner}/{repo} -L 1000 --state all"
-    pipe_command_to_file(command, output_file)
-
-
-def export_github_data():
-    print("Exporting issue data")
-    export_issue_data(
-        owner=OWNER,
-        repo=REPO,
-        output_file="data/issue-data.json",
-    )
-    print("Exporting sprint data")
-    export_project_data(
-        owner=OWNER,
-        project=SPRINT_PROJECT,
-        output_file="data/sprint-data.json",
+    """Export and write GitHub issue data to a JSON file."""
+    print(f"Exporting issue data from {owner}/{repo} to {output_file}")
+    command = (
+        f"gh issue list --json number,createdAt,closedAt,labels,title "
+        f"-R {owner}/{repo} -L 1000 --state all"
```

Comment:
I worry a 1000 issue max could cause secret problems for us later? Do we really need this, and if we do need it, can we make it clear and flag this so we're not surprised later?

Reply:
Good call! We do need to specify a max, otherwise the default is 50. But I can pull this out to a constant and set it much higher.

Reply:
Long-term, if we go the route of setting up a data warehouse I can also tweak this to only pull in issues that were modified or created in within a fixed range of time so that we don't have to export the entire repository every time.

Reply:
I pulled this out int a separate constant in [this commit](https://github.com/HHS/simpler-grants-gov/pull/782/commits/631060b0a3e168bf27534df680f6cae58888be92) on #782